Source/doom.cpp

```diff
@@ -44,48 +42,32 @@ int doom_get_frame_from_time()
 	return DoomQuestState / 1200;
 }
 
-void doom_cleanup()
-{
-	if (pDoomCel != nullptr) {
-		MemFreeDbg(pDoomCel);
-		pDoomCel = nullptr;
-	}
-}
-
-static bool doom_alloc_cel()
-{
-	doom_cleanup();
-	pDoomCel = DiabloAllocPtr(0x39000);
-	return pDoomCel != nullptr;
-}
-
 static bool doom_load_graphics()
 {
 	bool ret;
 
 	ret = false;
 	strcpy(tempstr, "Items\\Map\\MapZtown.CEL");
-	if (LoadFileWithMem(tempstr, pDoomCel) != 0)
+	if (LoadFileWithMem(tempstr, doomCel.get()) != 0)
 		ret = true;
 	return ret;
 }
 
 void doom_init()
 {
-	if (doom_alloc_cel()) {
-		doom_quest_time = doom_get_frame_from_time() == 31 ? 31 : 0;
-		if (doom_load_graphics()) {
-			doomflag = true;
-		} else {
-			doom_close();
-		}
+	doomCel = std::make_unique<uint8_t[]>(0x39000);
+	doom_quest_time = doom_get_frame_from_time() == 31 ? 31 : 0;
+	if (doom_load_graphics()) {
+		doomflag = true;
+	} else {
+		doom_close();
 	}
 }
 
 void doom_close()
 {
 	doomflag = false;
-	doom_cleanup();
+	doomCel.reset();

```

Comment:
nit: I find that this reads a bit better:

```suggestion
	doomCel = nullptr;
```

Reply:
would you apply this also?

Reply:
Sure, why not?